static/components.interfaces.ts

```diff
@@ -61,14 +61,14 @@ export interface ComponentConfig<S> {
 }
 
 type StateWithLanguage = {lang: string};
-type StateWithEditor = {source: number};
+type StateWithEditor = {source: string | number};
```

Comment:
This is set to a number in a couple of places (e.g. `source: editorId`)

Reply:
This looks more like a bug in that code that the type being wrong in this case. I think we should work towards normalizing state types. I _think_ all states end up with the id being either one or the other and we don't actually match, right? We should try to move all of them to there

Reply:
Agreed: perhaps a `TODO(#somenumber)` to come back and fix if it's not obvious how to proceed quickly (so this doesn't get stale).

Reply:
Thank you! Good points and suggestions